### PR TITLE
Fling

### PR DESCRIPTION
**Pull request by codejvn**

### CODE REVIEW

File: src/main/java/frc/robot/Superstructure.java [hunk 1]
```diff
@@ -334,6 +346,27 @@ public void periodic() {
...
+          double shooterSetPoint = shooterflingInterpolation.getValue(
+                  RobotContainer.poseEstimator.distanceBetweenPoses(
+                      RobotContainer.poseEstimator.getLatestPose(), flingPose));
+          arm.setPosition(armDistanceSetPoint);
+          shooter.shoot(shooterSetPoint);
+          if (arm.getState() == ArmStates.AT_SETPOINT
+              && shooter.getState() == ShooterStates.AT_SETPOINT) {
+            conveyor.setShooting();
+          }
+          break;
```
Comment (codejvn) [inline src/main/java/frc/robot/Superstructure.java:368]: should be outside of the isPresent() conditional

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -267,6 +267,11 @@ public void updateUserInput() {
     // }
     else if (input.controllerButton(7)) {
       superstructure.resetRobot();
+    } else if (input.rightJoystickButton(15)) {
+      DriveCommands.SnapToTarget(
+              drive, () -> -input.leftJoystickY(), () -> -input.leftJoystickX(), Targets.FLING)
+          .until(() -> input.rightJoystickButton(2));
```
Comment (codejvn) [inline src/main/java/frc/robot/RobotContainer.java:273]: the until should be the same button as the button the driver is pressing (aka it should be `.until(() -> input.rightJoystickButton(15));` )

**Review by codejvn — COMMENTED** [review]
looks good overall, just minor things

File: src/main/java/frc/robot/RobotContainer.java (hunk 2 above)
Comment (dmanslick) [reply src/main/java/frc/robot/RobotContainer.java]: Both of the issues should be fixed in the latest commit on fling

**Pull request merged**